### ISSUE TITLE
adaptivePtime property for RTCRtpEncodingParameters

### ISSUE THREAD

**Issue by jakobivarsson** [mozilla/standards-positions#508]
## Request for Mozilla Position on an Emerging Web Specification

* Specification Title: A new flag in RTCRtpEncodingParameters for adaptive packet rate
* Specification or proposal URL: https://w3c.github.io/webrtc-extensions/explainer#a-new-flag-in-rtcrtpencodingparameters-for-adaptive-packet-rate, https://w3c.github.io/webrtc-extensions/#dom-rtcrtpencodingparameters-adaptiveptime

### Other information

The flag is currently available through an origin trial in Chrome (https://chromestatus.com/feature/5752004691361792).

**Comment by martinthomson** [comment]
@nils-ohlmeier probably has the best shot at assessing this one, but that might need to wait until he is back from vacation.

**Comment by foolip** [comment]
@martinthomson would you happen to know when @nils-ohlmeier is back? I see that @jan-ivar was poked on https://github.com/w3c/webrtc-pc/issues/2300 which led up to this proposal, but didn't weigh in there.

**Comment by jakobivarsson** [comment]
We did get some feedback in: https://github.com/w3c/webrtc-pc/pull/2309#issuecomment-538463368

**Comment by jan-ivar** [comment]
Yes we did give feedback at the time that this "seems fine". I'd like to check with @nils-ohlmeier who should be back Monday, to make sure nothing has changed, before forming an official position though.

**Comment by yoavweiss** [comment]
Friendly ping! :)

**Comment by nils-ohlmeier** [comment]
In general I'm not opposed to the feature.

But I have a couple of concerns:
- https://w3c.github.io/webrtc-extensions/#dom-rtcrtpencodingparameters-adaptiveptime says that `adaptivePtime` is a read-only parameter. That doesn't make much sense to me.
- From https://w3c.github.io/webrtc-extensions/#dom-rtcrtpencodingparameters-adaptiveptime `If adaptivePtime is set to true, ptime MUST NOT be set...` which `ptime` does this refer to? I assume `ptime` in SDP, but that should be clarified.
- From https://w3c.github.io/webrtc-extensions/#dom-rtcrtpencodingparameters-adaptiveptime `otherwise, InvalidModificationError MUST be thrown.` what means `otherwise`? If an application tries to set this to `false`? Or if the browser doesn't support this feature?
- From looking at https://github.com/w3c/webrtc-pc/pull/2309 it appears that the agreement is to not signal this. If that is the agreement I think the spec needs stronger wording around this feature that the app is only suppose to use this if it knows that the receiver supports it. So basically we assume knowledge in the app again about supported features on the receiver side. Which is highly problematic if we consider use cases where different audio goes to different endpoints. I'm concerned that in worst case a call can start and the person on the receiver hears the audio. But when the browser decides to switch the frame rate suddenly the user won't hear anything any more. Until the browser decides to switch to something which the receiver can handle. I'm not convinced the easier and faster deployment arguments raised in the GitHub issue warrant bypassing signaling.
